src/metaswitch/clearwater/config_manager/config_access.py

```diff
@@ -343,19 +353,48 @@ def main(args):
         try:
             upload_verified_config(config_loader,
                                    local_store,
-                                   args.config_type,
+                                   selected_config,
                                    args.force,
                                    args.autoconfirm)
         except (UserAbort, ConfigUploadFailed) as exc:
             log.error("Upload failed")
             sys.exit(exc)
 
 
+def lookup_config_type(config_type, config_location):
+    """
+    This takes the string defining the config_type from arg.parse and returns
+    a instance of the class containing data on the config type selected
+    """
+    config_classes = load_plugins_in_dir(PLUGIN_DIR, config_location)
+    for choice in config_classes:
+        if config_type == choice.name:
+            return choice
```

Comment:
Last time, I mentioned the [next](https://docs.python.org/2/library/functions.html#next) keyword. If you're not going to handle the case where none of the plugins have the right name, then it can greatly simplify your code here.

Have a look at what it is supposed to do and see if you can figure out how to use it. Come and talk to me if you get stuck.